### PR TITLE
New Doc Link

### PR DESCRIPTION
**Pull request by DLu**
### Description

Necessary yak-shaving before #234. Adds new extlinks format for linking to [ROS 2 docs](https://github.com/ros2/ros2_documentation)

### Checklist
- [x] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] While waiting for someone to review your request, please consider reviewing [another open pull request](https://github.com/ros-planning/moveit2/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: doc/getting_started/getting_started.rst [hunk 1]
```diff
@@ -5,7 +5,7 @@ This tutorial will install MoveIt 2 and create a workspace sandbox to run the tu
 
 Install ROS 2 and Colcon
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-`Install ROS 2 Foxy <https://docs.ros.org/en/foxy/Installation.html>`_.
```
Comment (DLu) [inline doc/getting_started/getting_started.rst:8]: I could not for the life of me figure out how to get the DISTRO macro to resolve.

Comment (tylerjw) [reply doc/getting_started/getting_started.rst:8]: Me neither.  I know the ros2 docs use them and I copied them from there but I haven't figured out what we are doing wrong with them.  This is a reasonable solution for now as it is how we resolve many of our other links.

**Review by tylerjw — APPROVED** [review]

**Comment by tylerjw** [comment]
I does look like you need to run pre-commit (which will run black) to get it to pass the format CI job.

File: doc/getting_started/getting_started.rst (hunk 1 above)
Comment (DLu) [reply doc/getting_started/getting_started.rst:8]: The knowledge that ros2 docs do it was enough to point me in the right direction.

**Review by tylerjw — APPROVED** [review]

**Comment by DLu** [comment]
Build is broken on some hybrid planner shiz @henningkayser 

```
In file included from /home/runner/work/moveit2_tutorials/moveit2_tutorials/.work/upstream_ws/src/moveit2/moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/src/hybrid_planning_manager.cpp:35:
  /home/runner/work/moveit2_tutorials/moveit2_tutorials/.work/upstream_ws/src/moveit2/moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/include/moveit/hybrid_planning_manager/hybrid_planning_manager.h:44:10: fatal error: moveit_msgs/action/local_planner.hpp: No such file or directory
     44 | #include <moveit_msgs/action/local_planner.hpp>
        |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  compilation terminated.
```

**Comment by DLu** [comment]
Conflicts resolved, CI passes.

**Pull request merged**